### DISCUSSION TITLE
MY first data processor 0.92.0

### DISCUSSION THREAD

**Discussion by lirarl** [apache/streampipes#2210] [Q&A] [answered]
Hello 

I am trying to write my own new data processor according to instruction https://streampipes.apache.org/docs/extend-first-processor/
however I ma failing. I am  new with Apache Stream pipes and maybe you can help 

I set up environment on Windows like this 

$ streampipes env --set pipeline-element
[INFO] Set environment to 'pipeline-element'
$ streampipes up -d

When I try to run my Processor from IDE

![Capture](https://github.com/apache/streampipes/assets/47687271/1d0dfcb7-500f-4505-ba2f-01ee28f5410c)

Log:
2023-11-20T13:40:28.492+01:00  INFO   --- [           main] o.a.s.s.consul.ConsulProvider            : Checking if consul is available on host consul and port 8500
2023-11-20T13:40:30.779+01:00  INFO   --- [           main] o.a.s.s.consul.ConsulProvider            : Could not connect to Consul instance...
2023-11-20T13:40:30.779+01:00  INFO   --- [           main] o.a.s.s.consul.ConsulProvider            : Retrying in 1 second

When check in browser
'localhost:8090' 
I do not get anything

**Comment by bossenti** [comment, marked as answer]
Hi @lirarl,

great to hear that you're working on a custom StreamPipes processor 👏🏼

Did you pass any environment variables to the run configuration of your `Init` class?

**Reply by lirarl** [reply]
Thanks I forgot to add Environment Variables in IDE

After adding them
![image](https://github.com/apache/streampipes/assets/47687271/0a92068a-705c-4271-82f3-5d473509e93d)
It start Working

**Reply by bossenti** [reply]
Perfect!

If you have any additional questions or encounter any issues, please feel free to reach out. We're here to help! 🙂
